src/syntax/lexer.ml

```diff
@@ -277,6 +277,20 @@ let ident = [%sedlex.regexp?
 	)
 ]
 
+let sharp_ident = [%sedlex.regexp?
```

Comment:
@RealyUniqueName Could you double-check the regex? The idea is to match `part.part` but not `part.` or `part..part` or other weird stuff.

Reply:
`[a-z_]+(\.[a-zA-Z0-9_]+|[a-zA-Z0-9_]*)*`

This allows identifiers after the first one to begin with numerals or uppercase letters, e.g. `target.9` – do you want to allow that?

Reply:
How about

`[a-z_][a-zA-Z0-9_]*(\.[a-z_][a-zA-Z0-9_]*)*`

Reply:
Yes good point. Could you translate that to sedlex and push it?